### ISSUE TITLE
The client has not received the message yet but message already send to the dead letter topic.

### ISSUE THREAD

**Issue by codelipenghui** [apache/pulsar#3077]
#### Expected behavior

User process the same message many times but failed, if user set a dead letter policy, when message process times exceed the max redelivery count in dead letter policy, message will send to the dead letter topic.

#### Actual behavior

When a consumer subscribe a topic, but wait a while then start receive messages, but messages already send to dead letter topic.

#### Steps to reproduce

Here is the code to reproduce

```java
public class RedeliveryIssue {

    public static void main(String[] args) throws PulsarClientException, InterruptedException {

        final String topic = "my-topic";

        PulsarClient client = PulsarClient.builder()
                .serviceUrl("pulsar://localhost:6650")
                .build();

        Consumer<byte[]> consumer = client.newConsumer()
                .topic(topic)
                .subscriptionType(SubscriptionType.Shared)
                .subscriptionName(UUID.randomUUID().toString())
                .ackTimeout(3, TimeUnit.SECONDS)
                .deadLetterPolicy(DeadLetterPolicy.builder().maxRedeliverCount(2).build())
                .subscribe();

        Producer<byte[]> producer = client.newProducer()
                .topic(topic)
                .create();

        producer.send(("a message").getBytes());

        // wait a while, message will send to dead letter topic
        Thread.sleep(10000L);

        do {
            // can't receive message
            Message<byte[]> msg = consumer.receive();
            System.out.println(new String(msg.getValue()));
        } while (true);
    }
}
```

#### System configuration
**Pulsar version**: 2.2.0

**Comment by codelipenghui** [comment]
PR #3079 for fix this issue

**Comment by sijie** [comment]
This has been fixed by #3079
